### PR TITLE
Move key types to central location

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Moves key types to a central type file. I plan on expanding these opaque types to include asymmetric keys, which make sense in a single key type file rather than distributed between multiple.

All changes are either moving these opaque types or import updates.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Comment by JaredSnider-Bitwarden** [comment]
@MGibson1 , this is a great change, but looks like there are several build failures to address still.

**Review by djsmith85 — APPROVED** [review]

**Review by JaredSnider-Bitwarden — DISMISSED** [review]
LGTM!

**Review by Hinton — APPROVED** [review]

File: libs/common/src/types/key.d.ts
Comment (justindbaur) [inline libs/common/src/types/key.d.ts:1]: Why a `.d.ts` file? TypeScript maintainers aren't a fan of using them for this. https://github.com/microsoft/TypeScript/issues/52593#issuecomment-1419505081

**Review by Jingo88 — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by cagonzalezcs — DISMISSED** [review]

**Review by shane-melton — APPROVED** [review]

File: libs/common/src/types/key.d.ts
Comment (MGibson1) [reply libs/common/src/types/key.d.ts:1]: didn't know that, I'd always thought they were indicators of types only

**Review by shane-melton — APPROVED** [review]

**Review by cagonzalezcs — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

**Pull request merged**